### ISSUE TITLE
Deal with Notebook smoke tests

### ISSUE THREAD

**Issue by joaomoreno** [microsoft/vscode#126371]
The [entire Notebook smoke test suite](https://github.com/microsoft/vscode/blob/9d4274bf64e7f19881d8c6f35e2af3d273e41d3b/test/smoke/src/areas/notebook/notebook.test.ts) is currently entirely skipped. We should address this by either fixing the tests or removing them altogether.

cc @jrieken @roblourens

**Comment by rebornix** [comment]
I'll recommend us removing them.

**Comment by bpasero** [comment]
Related, they are entirely disabled for web in integration tests: https://github.com/microsoft/vscode/issues/126384

**Comment by roblourens** [comment]
Current status: I'd like to delete the ones that have just always been flaky but leave at least a couple so we open a notebook during smoketesting. But opening a notebook fails in the web smoketests due to something with the webview.

**Comment by roblourens** [comment]
@mjbvz can you look at this? I also can't do a web integration test for (I think) the same reason.

```
Error loading webview: Error: Could not register service workers: TypeError: Failed to register a ServiceWorker for scope ('https://a3772875-0e58-451e-80db-0d430bdac14d.vscode-webview-test.com/5319757634f77a050b49c10162939bfe60970c29/') with script ('https://a3772875-0e58-451e-80db-0d430bdac14d.vscode-webview-test.com/5319757634f77a050b49c10162939bfe60970c29/service-worker.js?id=a3772875-0e58-451e-80db-0d430bdac14d&swVersion=2&extensionId=&platform=browser&vscode-resource-base-authority=vscode-resource.vscode-webview.net&purpose=notebookRenderer'): An unknown error occurred when fetching the script..
```

Works like in yarn web though, I don't know what the difference it

**Comment by bpasero** [comment]
I would like to enrich our hot-exit smoke tests to also open a notebook, make it dirty and verify the contents are there after restart. But given the notebook tests are disabled, it looks I am blocked on this? What is the status?

**Comment by roblourens** [comment]
@mjbvz would you have time to look at this in December/Jan?

**Comment by roblourens** [comment]
These failed in the browser in CI and I see this error

```
index.html?id=c7636a08-435f-4f8c-928b-3fd04bc4768d&swVersion=3&extensionId=&platform=browser&vscode-resource-base-authority=vscode-resource.vscode-webview.net&parentOrigin=http%3A%2F%2Flocalhost%3A9002&purpose=notebookRenderer:0Failed to load resource: the server responded with a status of 404 ()
```

They work locally, which I think they didn't last I looked at this.

**Comment by mjbvz** [comment]
Thanks rob. I think I may understand the issue now. Testing again with private branch to confirm

**Comment by roblourens** [comment]
Thanks. Notebook integration tests have flaked a few times recently and have similar errors: 

```
    1) cell execute command takes arguments ICellRange[]
Request Failed https://9827e884-46d1-446f-8fb2-7f33d9de535e.vscode-webview.net/insider/69df0500a8963fc469161c038a14a39384d5a303/out/vs/workbench/contrib/webview/browser/pre/fake.html?id=2a2aeb17-bb3c-4c1b-bd17-1ddbbcfe1674 net::ERR_ABORTED

    1) cell execute command takes arguments ICellRange[]
Request Failed https://4ca915b8-11b5-4ad4-b986-fcd47bf026f1.vscode-webview.net/insider/69df0500a8963fc469161c038a14a39384d5a303/out/vs/workbench/contrib/webview/browser/pre/main.js net::ERR_ABORTED


    1) #98841, initialzation should not emit cell change events.
Request Failed https://c2242a7c-6935-41c2-84c0-c71d1651ddaa.vscode-webview.net/insider/69df0500a8963fc469161c038a14a39384d5a303/out/vs/workbench/contrib/webview/browser/pre/index.html?id=e9dbf5fb-d3d7-4035-b0e6-34d0dc67aac9&swVersion=3&extensionId=&platform=browser&vscode-resource-base-authority=vscode-resource.vscode-webview.net&parentOrigin=http%3A%2F%2Flocalhost%3A8000&purpose=notebookRenderer NS_BINDING_ABORTED
```

It looks like the notebook didn't open successfully, I assume it's related somehow, I will disable them against this issue

**Comment by mjbvz** [comment]
Reopening to investigate integration tests too

**Comment by rebornix** [comment]
Current status of notebook integration tests:

- [ ]  `notebook.api.test`
  - [ ] API suite skipped on Web
  - [ ] `showTextDocument` skipped https://github.com/microsoft/vscode/issues/123270
  - [ ] Statusbar suite skipped https://github.com/microsoft/vscode/issues/139324
- [ ] `notebook.kernel.test`
  - [ ] Whole suite skipped on Web
  - [ ] Flaky `asPromise`  and `notebook.execute`
- [x] `notebook.document.ts`
- [ ]  `notebook.editor.ts`
  - [ ] Whole suite skipped on Web
  - [x] Skipped tests

**Comment by MrRogerTheShrubber** [comment]
> The [entire Notebook smoke test suite](https://github.com/microsoft/vscode/blob/9d4274bf64e7f19881d8c6f35e2af3d273e41d3b/test/smoke/src/areas/notebook/notebook.test.ts) is currently entirely skipped. We should address this by either fixing the tests or removing them altogether.
> 
> cc @jrieken @roblourens

Wait, notebooks don't get tested??

Hi guys! I was thinking to myself, and I reached the conclusion that public shaming bad projects with bad maintainers is a reasonable way to entice action.

So I'll tag some organizations in particularly egregious examples of bad behavior with the hopes that given enough eyeball, all bugs are shallow!

I sure hope that one day the guys at @google, @aws, or even @facebook, maybe @apple, that someone, will provide a great open source competitor so that we don't get locked into using the bad @microsoft @python extension as the python experience! 😢

Of I course I must mention the @eclipse @apache and @mozilla guys lest they get jealous! ❤️

**Comment by MrRogerTheShrubber** [comment]
@github/open-source

**Comment by rebornix** [comment]
> Wait, notebooks don't get tested??

Thanks for your interest in this topic, but this is not true. We have unit tests, and integration tests for notebooks, and we do manual tests during endgame each iteration https://github.com/microsoft/vscode/wiki/Running-the-Endgame. Smoke test is one of the test suite we have, and it's not the most effective one for notebook thus when it was flaky and blocking our CI repeatedly, we have it disabled.

**Comment by MrRogerTheShrubber** [comment]
> > Wait, notebooks don't get tested??
> 
> Thanks for your interest in this topic, but this is not true. We have unit tests, and integration tests for notebooks, and we do manual tests during endgame each iteration https://github.com/microsoft/vscode/wiki/Running-the-Endgame. Smoke test is one of the test suite we have, and it's not the most effective one for notebook thus when it was flaky and blocking our CI repeatedly, we have it disabled.

Oh yeah, makes sense to disable tests that fail. 

Hi guys! I was thinking to myself, and I reached the conclusion that public shaming bad projects with bad maintainers is a reasonable way to entice action.

So I'll tag some organizations in particularly egregious examples of bad behavior with the hopes that given enough eyeball, all bugs are shallow!

I sure hope that one day the guys at @google, @aws, or even @facebook, maybe @apple, that someone, will provide a great open source competitor so that we don't get locked into using the bad @microsoft @python extension as the python experience! 😢

Of I course I must mention the @eclipse @apache and @mozilla guys lest they get jealous! ❤️

**Comment by rebornix** [comment]
I tried to enable web tests for notebook and saw these errors

```
✔ notebook open (1289ms)
Request Failed https://vscode-remote+localhost-003a9888.vscode-resource.vscode-cdn.net/mnt/vss/_work/1/s/extensions/notebook-renderers/renderer-out/index.js net::ERR_ABORTED

```

The tests pass and the failures might be due to we close the editor too quickly. @mjbvz can you help confirm?

**Comment by mjbvz** [comment]
Yes I think that's ok in this case. We cancel resource loads when the webview is closed which is likely why you see this error printed

**Comment by DonJayamanne** [comment]
Removing @mjbvz for now from this, as it seems there might no longer be an issue.
Will need to enable the tests and verify once again.
